### ISSUE TITLE
ressource creation failed, but added to state

### ISSUE THREAD

**Issue by ARZ-SagA** [dynatrace-oss/terraform-provider-dynatrace#36]
Hi!

I created the following ressource, and forgot to add the id, the api error was correct but the ressource was added to the state
```
resource "dynatrace_notification" "TO240-ZIS-Webhook" {
  web_hook {
    name = "TO240 - ZIS System Alerting"
    accept_any_certificate = true
    active = true
    alerting_profile = ""
    payload = "{\n\"ProblemID\":\"{ProblemID}\",\n\"Message\":\"{ProblemTitle}\",\n\"State\":\"{State}\",\n\"ProblemImpact\":\"{ProblemImpact}\",\n\"ProblemSeverity\":\"{ProblemSeverity}\",\n\"Impacted\":{ImpactedEntities},\n\"URL\":\"{ProblemURL}\",\n\"Tags\":\"{Tags}\"\n}"
    url = "demourl"
  }
}
```

```
dynatrace_notification.TO240-ZIS-Webhook: Creating...

Error: {
  "code": 400,
  "message": "Constraints violated.",
  "constraintViolations": [
    {
      "parameterLocation": "PAYLOAD_BODY",
      "location": "",
      "message": "may not be null",
      "path": "alertingProfile"
    }
  ]
}
```

it would be nice if the payload would be checked on terraform plan against the validator api

**Comment by Dynatrace-Reinhard-Pilz** [comment]
What's interesting here is, that terraform apparently doesn't refuse to add the configuration to the state, even if an error is being reported. Moreover, the state is being produced by sending a GET request for the resource that just has been created.
I will test out your sample and see what's going on under the hoods.
We were hoping to get away without Validator API unless the resource consists of more than one REST calls. I'll look into that for that specific case.

**Comment by Dynatrace-Reinhard-Pilz** [comment]
Hi @ARZ-SagA

The resource is indeed getting added to the state, but with status `tainted`. Essentially Terraform puts down a note that this resource is required to either get deleted or recreated during the next apply. 

Root cause here was a remainder from a debugging session. The ID of the resource was set to a dummy value before an attempt to create the resource could have been made.

Checking the payload against the validator API wouldn't have helped here. To be more specific, I haven't found a case yet, where an extra call to the validator API would have resulted in a different status then allowing the endpoint for creating the resource performing the validation right away.

Your issue will be fixed with v1.2.0
